lib/test.js

```diff
@@ -20,12 +20,13 @@ module.exports = Test;
  * @api private
  * @param {String} title
  * @param {Function} fn
+ * @param {String} reason
```

Comment:
Please revert changes to this constructor.  Even if `reason` must be a property of a `Test` or `Runnable` (I'm not entirely convinced), it should be renamed `pendingReason` and settable/gettable via methods.

Please investigate using the reason as the `message` of a thrown `Pending` object instead.  I see at least one place where the `Runner` emits `pending` where we have access to the `Pending` instance itself... that might be all we need.  Then, we can do something like `this.emit('pending', test, pending.message)` and the reporter can handle it.